app/controllers/manageBusinesses/manage/CheckYourAnswersController.scala

```diff
@@ -0,0 +1,226 @@
+                                           val sessionService: SessionService,
+                                           val auditingService: AuditingService,
+                                           val auth: AuthenticatorPredicate)
+                                          (implicit val ec: ExecutionContext,
+                                           implicit override val mcc: MessagesControllerComponents,
+                                           implicit val itvcErrorHandler: ItvcErrorHandler,
+                                           implicit val itvcErrorHandlerAgent: AgentItvcErrorHandler,
+                                           val appConfig: FrontendAppConfig) extends ClientConfirmedController
+  with FeatureSwitching with IncomeSourcesUtils with JourneyCheckerManageBusinesses {
+
+  def show(isAgent: Boolean, incomeSourceType: IncomeSourceType): Action[AnyContent] = auth.authenticatedAction(isAgent) { implicit user =>
+    withSessionData(JourneyType(Manage, incomeSourceType), journeyState = BeforeSubmissionPage) { sessionData =>
+      val incomeSourceIdStringOpt = sessionData.manageIncomeSourceData.flatMap(_.incomeSourceId)
+      val chnageToStringOpt = sessionData.manageIncomeSourceData.flatMap(_.reportingMethod)
+      val taxYearStringOpt = sessionData.manageIncomeSourceData.flatMap(_.taxYear)
+      val incomeSourceIdOpt = incomeSourceIdStringOpt.map(id => IncomeSourceId(id))
+      handleShowRequest(taxYearStringOpt,
+        chnageToStringOpt,
+        isAgent,
+        incomeSourceType,
+        incomeSourceIdOpt,
+        backUrl = {
+          if (isAgent) controllers.routes.HomeController.showAgent
+          else controllers.routes.HomeController.show()
+        }.url
+      )
+    }
+  }
+
+  private def handleShowRequest(taxYearStringOpt: Option[Int], chnageToStringOpt: Option[String], isAgent: Boolean, incomeSourceType: IncomeSourceType, incomeSourceIdOpt: Option[IncomeSourceId], backUrl: String)(implicit user: MtdItUser[_]): Future[Result] = {
+    val maybeIncomeSourceId: Option[IncomeSourceId] = user.incomeSources.getIncomeSourceId(incomeSourceType, incomeSourceIdOpt.map(m => m.value))
+    withIncomeSourcesFS {
+      Future.successful(
+        (taxYearStringOpt, chnageToStringOpt, maybeIncomeSourceId) match {
+          case (Some(taxYearStringOpt), Some(changeToStringOpt), Some(id)) => {
+            val taxYearStartYear = taxYearStringOpt - 1
+            val taxYearEndYear = taxYearStringOpt
+            Ok(checkYourAnswers(
+              isAgent,
+              backUrl,
+              CheckYourAnswersViewModel(id, changeToStringOpt, taxYearStartYear.toString, taxYearEndYear.toString, incomeSourceType),
+              incomeSourceType.journeyType
+            )
+            )
+          }
+          case (None, _, _) => logAndShowError(isAgent, s"[handleShowRequest]: Could not parse taxYear: $taxYearStringOpt")
```

Comment:
you can reduce these 3 error case to case (_, _, _) and provide failed data in the log message